### PR TITLE
fix: stop genesis validators on fork

### PR DESCRIPTION
**Pull request by AKorpusenko**
## Description:

- Added primitive `ForkMonitor()` which is checks is fork is passed or not
- In case it passed the fork epoch, then `CleanGenesisValidators()` is called to stop all genesis validators instances
- Added skipping the initialization of `genesisValidator` in `onShareInit()`

### CODE REVIEW

**Comment by moshe-blox** [comment]
made a refactor to stop all validators at once with a parent context, and a bunch of other changes for nil safety and memory freeing

**Review by y0sher — APPROVED** [review]
lgtm

**Review by AKorpusenko — COMMENTED** [review]
@moshe-blox, lgtm 👍

**Comment by y0sher** [comment]
@AKorpusenko please test this in `stage` ON-fork to see it works and doesn't break.

File: operator/validator/controller.go [hunk 1]
```diff
@@ -1242,38 +1256,31 @@ func (c *controller) startCommittee(vc *validator.Committee) (bool, error) {
...
-		case <-c.context.Done():
-			return
-		case <-time.After(timeUntilFork):
-			logger.Info("Stopping genesis validators after alan fork")
-			c.CleanGenesisValidators()
-			logger.Info("✅Successfully stopped genesis validators")
+		case <-c.ctx.Done():
 			return
+		case <-next:
+			next = slotTicker.Next()
```
Comment (AKorpusenko) [inline operator/validator/controller.go:1281]: @moshe-blox, do we really need this?

File: operator/validator/controller.go [hunk 2]
```diff
@@ -1242,38 +1256,31 @@ func (c *controller) startCommittee(vc *validator.Committee) (bool, error) {
...
 
+	next := slotTicker.Next()
 	for {
 		select {
-		case <-c.context.Done():
-			return
-		case <-time.After(timeUntilFork):
-			logger.Info("Stopping genesis validators after alan fork")
-			c.CleanGenesisValidators()
-			logger.Info("✅Successfully stopped genesis validators")
```
Comment (AKorpusenko) [inline operator/validator/controller.go]: @moshe-blox, why did you remove the logs here? It happens once and bring some confidence stopping did happen

Comment (moshe-blox) [reply operator/validator/controller.go]: agree, i returned the log

File: operator/validator/controller.go (hunk 1 above)
Comment (moshe-blox) [reply operator/validator/controller.go:1281]: we don't really, but i don't trust very long sleeps to be precise

when we release Alan, the fork epoch could be set to around 2 weeks ahead, and i can't trust that if we schedule a Go timer for 2 weeks from now, it would fire precisely at the exact millisecond we need it to

given that system clocks often correct themselves, in such a long timeframe, i don't know what would be the impact of that

**Review by moshe-blox — APPROVED** [review]
